src/background/actions/injected-dialog-action-creator.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { TelemetryEventHandler } from 'background/telemetry/telemetry-event-handler';
+import * as TelemetryEvents from 'common/extension-telemetry-events';
+import { getStoreStateMessage, Messages } from 'common/messages';
+import { StoreNames } from 'common/stores/store-names';
+import { Interpreter } from '../interpreter';
+import { InjectedDialogOpenPayload } from './action-payloads';
+import { InjectedDialogActions as InjectedDialogActions } from './injected-dialog-actions';
+
+export class InjectedDialogActionCreator {
+    constructor(
+        private readonly interpreter: Interpreter,
+        private readonly injectedDialogActions: InjectedDialogActions,
+        private readonly telemetryEventHandler: TelemetryEventHandler,
+    ) {}
+
+    public registerCallbacks(): void {
+        this.interpreter.registerTypeToPayloadCallback(
+            Messages.InjectedDialog.Open,
+            this.onOpenDialog,
+        );
+        this.interpreter.registerTypeToPayloadCallback(
+            Messages.InjectedDialog.Close,
+            this.onCloseDialog,
+        );
+        this.interpreter.registerTypeToPayloadCallback(
+            getStoreStateMessage(StoreNames.InjectedDialogStore),
+            () => this.injectedDialogActions.getCurrentState.invoke(null),
+        );
+    }
+
+    private onOpenDialog = (payload: InjectedDialogOpenPayload): void => {
+        this.telemetryEventHandler.publishTelemetry(TelemetryEvents.ISSUES_DIALOG_OPENED, payload);
```

Comment:
Should we update the name for this telemetry event to `INJECTED_DIALOG_OPENED` for consistency? I also see the argument that we don't want to rename it because we already have a history of telemetry for `ISSUES_DIALOG_OPENED`, and we want them to be grouped together.

Reply:
Yep, that argument is why I left it as-is.